vunit/sim_if/nvc.py

```diff
@@ -299,14 +315,44 @@ def simulate(self, output_path, test_suite_name, config, elaborate_only):  # pyl
         except Process.NonZeroExitCode:
             status = False
 
+        if config.sim_options.get(self.name + ".gtkwave_script.gui", None):
+            warnings.warn(f"{self.name}.gtkwave_script.gui is deprecated and will be removed "
+                          f"in a future version, use {self.name}.viewer_script.gui instead")
+
         if self._gui and not elaborate_only:
-            cmd = ["gtkwave"] + shlex.split(self._gtkwave_args) + [str(wave_file)]
+            cmd = [self._get_viewer(config)] + shlex.split(self._viewer_args) + [str(wave_file)]
 
-            init_file = config.sim_options.get(self.name + ".gtkwave_script.gui", None)
+            init_file = config.sim_options.get(
+                self.name + ".viewer_script.gui",
+                config.sim_options.get(self.name + ".gtkwave_script.gui", None))
             if init_file is not None:
                 cmd += ["--script", str(Path(init_file).resolve())]
 
             stdout.write(f'{" ".join(cmd)}\n')
             subprocess.call(cmd)
 
         return status
+
+    def _get_viewer(self, config):
+        """
+        Determine the waveform viewer to use.
+
+        Falls back to gtkwave or surfer, in that order, if none is provided or the provided
+        cannot be found.
+        """
+        viewer = self._viewer or config.sim_options.get(self.name + ".viewer.gui", None)
+
+        if viewer is None and not self._gtkwave_available and not self._surfer_available:
```

Comment:
I think these two if statements should be replaced by this:

```
        if viewer is None:
            if self._gtkwave_available:
                viewer = "gtkwave"
            elif self._surfer_available:
                viewer = "surfer"
            else:
                raise RuntimeError("No viewer found. GUI not possible. Install GTKWave or Surfer.")

        elif not self.find_executable(viewer):
            raise RuntimeError(
                f"Cannot find the {viewer} executable in the PATH environment variable. GUI not possible"
            )
```

One reason is that if `viewer` is `None`, the second if statement will fail since the `find_executable` method expects a string. The other reason is that I think we should fail immediately if the given viewer can't be found. If we set the viewer to `smurfer` it's better to fail fast and fix the typo rather than being surprised by GTKWave popping up. Optionally, we can also print the available viewers in the error message.

Reply:
Good thinking!